src/fastga_he/gui/power_train_network_viewer.py

```diff
@@ -57,65 +187,437 @@ def power_train_network_viewer(
-            target = connection[1]
+            cleaned_node_types.append(_string_clean_up(node_type))
 
-        graph.add_edge(source, target)
+        cleaned_node_om_types.append(_string_clean_up(node_om_type))
+
+    # Define list info
+    hover_source = ColumnDataSource(
+        data=dict(
+            x=node_x,
+            y=node_y,
+            w=[s * icon_width_factor for s in node_sizes_list],
+            h=[s for s in node_sizes_list],
+            name=node_indices,
+            type_class=cleaned_node_types,
+            component_type=cleaned_node_om_types,
+        )
+    )
 
-    # The drawback of this method is that the creation of directories and file is not controled,
-    # it is created based on the working directory. So we will have to do some shenanigans to get
-    # and change the working directory once we are done with the creation of the graph.
+    # Add invisible circles on top for hover interactivity
+    plot.scatter(
+        x="x",
+        y="y",
+        size=60,
+        source=hover_source,
+        fill_alpha=0,
+        line_alpha=0,
+        hover_fill_alpha=0.1,
+        hover_line_alpha=0.3,
+    )
 
-    directory_to_save_graph = os.path.dirname(network_file_path)
-    graph_name = os.path.basename(network_file_path)
-    old_working_directory = os.getcwd()
+    hover = HoverTool(
+        tooltips=[
+            ("Name", "@name"),
+            ("Type class", "@type_class"),
+            ("Component type", "@component_type"),
+        ]
+    )
+    plot.add_tools(hover, BoxSelectTool())
 
-    # Set the new working directory and create it if it doesn't exist
-    if not os.path.exists(directory_to_save_graph):
-        os.makedirs(directory_to_save_graph)
+    return plot, edge_source, node_source, node_image_sequences, propeller_rotation_sequences
+
+
+def _add_color_legend_separate(plot, legend_position, color_icon_urls):
+    """
+    Add a color legend as a separate visual entity within the same plot.
+
+    Args:
+        plot: Bokeh figure object
+        legend_position: String defines the legend box position
+        (T: top, M: middle (vertical), Button, L: left, R: right, C: center (horizontal))
+        * * T * *
+        * * * * *
+        * * M * *
+        L * C * R
+        * * B * *
+        color_icon_urls: List of URLs for color icons
+    """
+
+    if "T" in legend_position:
+        legend_y_start = 600
+    elif "B" in legend_position:
+        legend_y_start = 50
+    elif "M" in legend_position:
+        legend_y_start = 325
+
+    if "R" in legend_position:
+        legend_x_start = 500
+    elif "L" in legend_position:
+        legend_x_start = -50
+    elif "C" in legend_position:
+        legend_x_start = 225
+
+    # Draw legend background box using quad
+    legend_box_source = ColumnDataSource(
+        data=dict(
+            left=[legend_x_start],
+            right=[legend_x_start + 100],
+            top=[legend_y_start],
+            bottom=[legend_y_start - 100],
```

Comment:
So 100 is the legend width and height ? Wouldn't it be better if we could change it ?